### PR TITLE
Add a grid decimation filter

### PR DESCRIPTION
**Pull request by alavenant**
Add a new filter who permit to keep the highest or lowest point in cells on a grid. It can help to keep the highest points of a forest, in order to have the canopy.

### CODE REVIEW

File: doc/stages/filters.griddecimation.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
+.. _filters.crop:
```
Comment (chambbj) [inline doc/stages/filters.griddecimation.rst:1]:
```suggestion
.. _filters.griddecimation:
```

File: doc/stages/filters.griddecimation.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+filters.crop
```
Comment (chambbj) [inline doc/stages/filters.griddecimation.rst:3]:
```suggestion
filters.griddecimation
```

File: doc/stages/filters.griddecimation.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+============
+
+The **grid decimation filter** keeps only one point in a grid. The point selected could be the highest or the lowest point. It can be used to quickly filter vegetation points in order to keep only the canopy points.
+
+.. embed::
+
+Example
+-------
+This example transform highest points of classification 5 in classification 9, on a grid of 0.75m square.
```
Comment (chambbj) [inline doc/stages/filters.griddecimation.rst:12]: In the description,  it says keep only the highest or lowest points. Why are they being classified to `value` then?

**Review by chambbj — COMMENTED** [review]
It is a bit cumbersome, but you can also use `filters.splitter` -> `filters.locate` -> `filters.merge` to achieve the same (assuming you are exporting and not classifying the points).

File: doc/stages/filters.griddecimation.rst (hunk 3 above)
Comment (alavenant) [reply doc/stages/filters.griddecimation.rst:12]: The description is not clear. My purpose is not to keep or delete points, but to transform (with the value) certains targeting points. We could, for exemple, add our own attribute who permits to select points for DSM (canopy for forest, lowest points for ground, etc...). I'm going to update the documentation.

**Comment by alavenant** [comment]
Hello, 
The macos conda integration doesn't work. It doesn't seem to come from my code. 
Maybe restart the test ? 
Antoine

**Comment by hobu** [comment]
Please rebase master or give maintainers the ability to merge to your branch. I would like to get this ready to include for PDAL 2.7.0, which is due by the end of the month.

**Comment by chambbj** [comment]
LGTM! Thanks @alavenant !!

**Pull request merged**